rust/src/morton.rs

```diff
@@ -4,45 +4,43 @@ use {Box3, Result, Vec3};
 #[derive(PartialEq, Debug)]
 pub struct Morton(u64);
 
-fn shuffle(v: u64) -> u64 {
-    // take first 21 bits
-    let mut z = v & 0x00000000001fffff;
-    z = (z | (z << 32)) & 0x001f00000000ffff;
-    z = (z | (z << 16)) & 0x001f0000ff0000ff;
-    z = (z | (z << 8)) & 0x100f00f00f00f00f;
-    z = (z | (z << 4)) & 0x100f00f00f00f00f;
```

Comment:
Here's the bug: The constant should be 0x10c30c30c30c30c3.
See https://github.com/Forceflow/libmorton/blob/24b63e34b1b3b5591fd10575c89ef1e5d1bb3e2a/libmorton/morton3D.h#L229

Reply:
:+1: 
This is the relevant issue https://github.com/Forceflow/libmorton/issues/25 and here's the relevant commit: https://github.com/Forceflow/libmorton/commit/d0d8c2c85dd16cec27a2703c8e89623c9a082503